localstack-core/localstack/services/transcribe/provider.py

```diff
@@ -235,11 +240,26 @@ def download_model(name: str):
 
             LOG.debug("Downloading language model: %s", model_path.name)
 
+            def _try_download(url: str):
+                try:
+                    download(url + str(model_path.name) + ".zip", model_zip_path, verify_ssl=False)
+                except Exception as e:
+                    LOG.error("Download failed from %s: %s", url, e)
+                    return False
+                return True
+
             from vosk import MODEL_PRE_URL  # noqa
 
-            download(
-                MODEL_PRE_URL + str(model_path.name) + ".zip", model_zip_path, verify_ssl=False
-            )
+            download_urls = [VOSK_MODEL_ARTIFACTS_URL, MODEL_PRE_URL]
+
+            for url in download_urls:
+                if _try_download(url):
+                    # Download the models from the localstack artifacts if present else download from the Vosk URL
+                    LOG.info("Successfully downloaded model from %s", url)
+                    break
+                else:
+                    LOG.error("Failed to download model from %s", url)
+                    return
```

Comment:
This pattern is a little hard to follow. Can I suggest simplifying this as follows:

```suggestion
            for url in download_urls:
                try:
                    download(
                MODEL_PRE_URL + str(model_path.name) + ".zip", model_zip_path, verify_ssl=False
            )
                except Exception as exc:
                    LOG.warning(...)
                    continue
                break
```